src/services/ghost/PromptStrategyManager.ts

```diff
@@ -0,0 +1,118 @@
+import { GhostSuggestionContext } from "./types"
+import { PromptStrategy } from "./types/PromptStrategy"
+import { ContextAnalyzer } from "./ContextAnalyzer"
+
+// Import all strategies
+import { UserRequestStrategy } from "./strategies/UserRequestStrategy"
+import { ErrorFixStrategy } from "./strategies/ErrorFixStrategy"
+import { SelectionRefactorStrategy } from "./strategies/SelectionRefactorStrategy"
+import { CommentDrivenStrategy } from "./strategies/CommentDrivenStrategy"
+import { NewLineCompletionStrategy } from "./strategies/NewLineCompletionStrategy"
+import { InlineCompletionStrategy } from "./strategies/InlineCompletionStrategy"
+import { AutoTriggerStrategy } from "./strategies/AutoTriggerStrategy"
+
+/**
+ * Manages prompt strategies and selects the appropriate one based on context
+ */
+export class PromptStrategyManager {
+	private strategies: PromptStrategy[]
+	private contextAnalyzer: ContextAnalyzer
+	private debug: boolean
+
+	constructor(debug: boolean = false) {
+		this.debug = debug
```

Comment:
Same annoying comment here- let's make this an object `{ debug }`
```suggestion
	constructor({ debug }: { debug: boolean }) {
		this.debug = debug ?? false
```